### PR TITLE
Add ipv6 validation

### PR DESCRIPTION
**Pull request by suramrit**
This pull requests addresses the following feature: 
 - ipv6 validation  and test coverage for ipv6 addressing 

👀  Seeking design approach from the repo contributors.

[WIP] 
🔧  golinter criteria fail for this current work due to the nature of all the edge cases that need to be considered for ipv6 which add to the size of the function definition
🤔  Proposed workaround - relax function size and line size to accommodate the edge cases

### CODE REVIEW

**Comment by patil-ashutosh** [comment]
@suramrit   This is great!  Thank your for raising your pull request.

**Comment by patil-ashutosh** [comment]
> This pull requests addresses the following feature:
> 
>     * ipv6 validation  and test coverage for ipv6 addressing
> 
> 
> 👀 Seeking design approach from the repo contributors.
> 
> [WIP]
> 🔧 golinter criteria fail for this current work due to the nature of all the edge cases that need to be considered for ipv6 which add to the size of the function definition
> 🤔 Proposed workaround - relax function size and line size to accommodate the edge cases

@suramrit Thanks for pointing this out, we can relax function size and line size to accommodate the edge cases. 
But I think you can fix golinter comment in ipv6Validation_test.go file (line number 63). please let me know if you need any help

File: regex/rvalidation/ipv6Validation_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+				t.Errorf("ValidateIpv6(%v) Failed: expected %t, actual %t",
+					tc.address, tc.expectedValidity, isValidIpv6)
+			}
+		})
+	}
+}
+
+func ExampleValidateIpv6() {
+	fmt.Println(rvalidation.ValidateIpv6("::1"))
+	//Output:true
```
[the remark this answers is not in the document]
Comment (suramrit) [reply regex/rvalidation/ipv6Validation_test.go:63]: Thanks for pointing this out!

**Comment by suramrit** [comment]
Thanks for the feedback! 😊 @patil-ashutosh . I will fix the _test file as you mentioned. 
 I also have another approach where I just loop through the different patterns, that could reduce the function definition. I'll push these changes today.
